app/experimenter/nimbus-ui/src/components/FormAudience/index.tsx

```diff
@@ -2,76 +2,178 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React from "react";
+import React, { useCallback } from "react";
+import {
+  useForm,
+  Controller,
+  ValidationRules,
+  FieldError,
+} from "react-hook-form";
 import Select from "react-select";
 import Form from "react-bootstrap/Form";
+import Alert from "react-bootstrap/Alert";
 import InputGroup from "react-bootstrap/InputGroup";
 import Col from "react-bootstrap/Col";
 import LinkExternal from "../LinkExternal";
 import InlineErrorIcon from "../InlineErrorIcon";
+import { useConfig } from "../../hooks/useConfig";
 
 import { getExperiment_experimentBySlug } from "../../types/getExperiment";
-import {
-  getConfig_nimbusConfig,
-  getConfig_nimbusConfig_channels,
-} from "../../types/getConfig";
+import { getConfig_nimbusConfig_channels } from "../../types/getConfig";
 
 // TODO: find this doco URL
 const AUDIENCE_DOC_URL =
   "https://mana.mozilla.org/wiki/pages/viewpage.action?spaceKey=FJT&title=Project+Nimbus";
 
-type FormAudienceConfig = Pick<
-  getConfig_nimbusConfig,
-  "channels" | "firefoxMinVersion" | "targetingConfigSlug"
->;
-
 export const FormAudience = ({
   experiment,
-  config,
+  submitErrors,
   isMissingField,
+  isLoading,
+  onSubmit,
+  onNext,
 }: {
   experiment: getExperiment_experimentBySlug;
-  config: FormAudienceConfig;
+  submitErrors: Record<string, string[]>;
   isMissingField: (fieldName: string) => boolean;
+  isLoading: boolean;
+  onSubmit: (data: Record<string, any>, reset: Function) => void;
+  onNext?: (ev: React.FormEvent) => void;
 }) => {
-  const {
-    channels,
-    firefoxMinVersion,
-    targetingConfigSlug,
-    populationPercent,
-    totalEnrolledClients,
-    proposedEnrollment,
-    proposedDuration,
-  } = experiment;
+  const config = useConfig();
```

Comment:
For some reason, when I first wrote FormAudience, I shied away from using useConfig here. I think I thought it would be hard to test? It wasn't, so I just switched over to pulling in config here for select field options

Reply:
FWIW,  I can select invalid channel options for the application and see a "Invalid channels for experiment application" here but that'll no longer be an issue after #4123 is in.